File: src/data-service/src/models/alpaca.py
from pydantic import BaseModel

from src.models.price import JsonDecimal


class AlpacaOrder(BaseModel):
    order_id: str
    ticker: str
    side: str  # "buy" or "sell"
    qty: JsonDecimal
    filled_avg_price: JsonDecimal
    filled_at: str  # ISO 8601 datetime

File: snapshots/chapter_4/backend/internal/model/alpaca.go
package model

type AlpacaOrder struct {
	OrderID        string  `json:"order_id"`
	Ticker         string  `json:"ticker"`
	Side           string  `json:"side"`
	Qty            float64 `json:"qty"`
	FilledAvgPrice float64 `json:"filled_avg_price"`
	FilledAt       string  `json:"filled_at"`
}

type ImportResult struct {
	Created int `json:"created"`
	Updated int `json:"updated"`
	Total   int `json:"total"`
}
